infra/feast-operator/api/v1alpha1/featurestore_types.go

```diff
@@ -499,6 +499,9 @@ type ServerConfigs struct {
 	// required by the Feast components. Ensure that each volume mount has a corresponding
 	// volume definition in the Volumes field.
 	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+	// Mode specifies the registry API serving mode.
+	// Allowed values: "grpc", "rest".
+	Mode string `json:"mode,omitempty"`
```

Comment:
is this specific to the registry? if so, you'll want to create a new `RegistryServerConfigs` struct with `ServerConfigs` inline.

also, thoughts on a bool instead? unless there was a reason you went w/ a `string` field?
```suggestion
	// Mode specifies the registry API serving mode.
	// Allowed values: "grpc", "rest".
	RestMode bool `json:"restMode,omitempty"`
```

Reply:
Updated!